avm/ptn/app-service-lza/hosting-environment/modules/front-door/approve-afd-pe.module.bicep

```diff
@@ -0,0 +1,83 @@
+@description('Optional. The location to deploy the Redis cache service.')
+param location string
+
+@description('Default value is OK. Sets how the deployment script should be forced to execute even if the script resource has not changed. Can be current time stamp')
+param utcValue string = utcNow()
+
+@description('Optional. The name of the user-assigned identity to be used to auto-approve the private endpoint connection of the AFD. Changing this forces a new resource to be created.')
+param idAfdPeAutoApproverName string = guid(resourceGroup().id, 'userAssignedIdentity')
+
+@description('Optional. The tags to be assigned to the created resources.')
+param tags object = {}
+
+var roleAssignmentName = guid(resourceGroup().id, 'contributor')
+var contributorRoleDefinitionId = resourceId(
+  'Microsoft.Authorization/roleDefinitions',
+  'b24988ac-6180-42a0-ab88-20f7382dd24c'
+)
+var deploymentScriptName = 'runAfdApproval'
+
+var uami = resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', idAfdPeAutoApproverName)
+
+@description('The User Assigned Managed Identity that will be given Contributor role on the Resource Group in order to auto-approve the Private Endpoint Connection of the AFD.')
+module userAssignedIdentity 'br/public:avm/res/managed-identity/user-assigned-identity:0.4.0' = {
+  name: '${uniqueString(deployment().name, location)}-uami'
+  params: {
+    name: idAfdPeAutoApproverName
+    location: location
+    enableTelemetry: true
```

Comment:
Beware, this must be passed as a parameter from the invoking template (e.g., deploy-spoke)